tests/test_search.py

```diff
@@ -932,6 +933,73 @@ def test_coadd_filter_gpu(self):
         self.assertEqual(meanStamps[2].width, 1)
         self.assertEqual(meanStamps[2].height, 1)
 
+    @staticmethod
+    def result_hash(res):
+        return hash((res.x, res.y, res.vx, res.vy, res.lh, res.obs_count))
+
+    def test_search_batch(self):
+        width = 50
+        height = 50
+        results_per_pixel = 8
+        min_observations = 2
+
+        # Simple average PSF
+        psf_data = np.zeros((5, 5), dtype=np.single)
+        psf_data[1:4, 1:4] = 0.1111111
+        p = PSF(psf_data)
+
+        # Create a stack with 10 20x20 images with random noise and times ranging from 0 to 1
+        count = 10
+        imlist = [make_fake_layered_image(width, height, 5.0, 25.0, n / count, p) for n in range(count)]
+        stack = ImageStack(imlist)
+        im_list = stack.get_images()
+        # Create a new list of LayeredImages with the added object.
+        new_im_list = []
```

Comment:
You don't need to create a new `ImageStack`. You should be able to insert the object directly into `stack`:

```
for i in range(count):
    im = stack.get_single_image(i)
    time = stack.get_zeroed_time(i)
    add_fake_object(im, 5.0 + (time * 8.0), 35.0 + (time * 0.0), 25000.0)
```

Reply:
Changed